  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

  wheel_left_rear_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: wheel_left_rear_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  wheel_right_rear_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: wheel_right_rear_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}

  steering_left_front_position_controller:
    type: position_controllers/JointPositionController
    joint: steering_left_front_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}
  steering_right_front_position_controller:
    type: position_controllers/JointPositionController
    joint: steering_right_front_joint
    pid: {p: 100.0, i: 0.01, d: 10.0}


      #   autoware:
      # joint_state_controller:
      #   type: joint_state_controller/JointStateController
      #   publish_rate: 50  
      
      # diff_drive_controller:
      #   type        : "diff_drive_controller/DiffDriveController"
      #   left_wheel  : 'wheel_left_rear_joint'
      #   right_wheel : 'wheel_right_rear_joint'
      #   publish_rate: 50.0               # default: 50
      #   pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
      #   twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

      #   # Wheel separation and diameter. These are both optional.
      #   # diff_drive_controller will attempt to read either one or both from the
      #   # URDF if not specified as a parameter
      #   wheel_separation : 1.55
      #   wheel_radius : 0.341

      #   # Wheel separation and radius multipliers
      #   wheel_separation_multiplier: 1.0 # default: 1.0
      #   wheel_radius_multiplier    : 1.0 # default: 1.0

      #   # Velocity commands timeout [s], default 0.5
      #   cmd_vel_timeout: 1.0
          
      #   # Base frame_id
      #   base_frame_id: base_link #default: base_link
          
      #   # Velocity and acceleration limits
      #   # Whenever a min_* is unspecified, default to -max_*
      #   linear:
      #     x:
      #       has_velocity_limits    : true
      #       max_velocity           : 0.55  # m/s
      #       min_velocity           : -0.55 # m/s
      #       has_acceleration_limits: true
      #       max_acceleration       : 1.0  # m/s^2
      #       min_acceleration       : -1.0 # m/s^2
      #   angular:
      #     z:
      #       has_velocity_limits    : true
      #       max_velocity           : 0.3  # rad/s
      #       min_velocity           : -0.3
      #       has_acceleration_limits: true
      #       max_acceleration       : 0.3  # rad/s^2
      #       min_acceleration       : -0.3
